### ISSUE TITLE
Wifi-connect does not work if I set a password (Raspbian)

### ISSUE THREAD

**Issue by antonmolodykh** [resin-io/resin-wifi-connect#158]
Hi guys.
I install wifi-connect on Raspbian using your [script](https://github.com/resin-io/resin-wifi-connect/blob/master/scripts/raspbian-install.sh)
This works fine, but only if I don't try to set a password. 
```
$ sudo wifi-connect -p test123
WiFi device: wlan0
Access points: ["GreatFruit", "OZPP", "GreatFruit", "sibalt", "YOTA", "Рейко", "Rosberry_2.4", "GROUP", "TP-LINK_D45E", "halif", "TP-LINK_D45E", "BSWifi"]
Starting access point...
Access point 'WiFi Connect' created
Starting HTTP server on 192.168.42.1:80
[wifi_connect:ERROR] Cannot start HTTP server on '192.168.42.1:80': address not available

dnsmasq: unknown interface wlan0
```
What do you think about this?

**Comment by majorz** [comment]
A second instance of WiFi Connect running at the same time possibly?

**Comment by antonmolodykh** [comment]
@majorz no. I checked everything many times, even reinstalled Raspbian. the problem exists

**Comment by majorz** [comment]
Thanks, I will try reproducing locally, see what happens and get back to you.

**Comment by majorz** [comment]
@antonmolodykh I installed a fresh Raspbian Lite 2017-11-29 on a RPi3 using the built-in WiFi interface. Installed latest WiFi Connect version and everything works as expected when adding the `-p` cli argument.

Can you please provide more details so that we see what is different between our setups?

**Comment by antonmolodykh** [comment]
@majorz I checked again. Steps to reproduce (on a Pi3 Model B):
1. Download and install [Raspbian Stretch](https://downloads.raspberrypi.org/raspbian_lite_latest)
2. Run the [script](https://github.com/resin-io/resin-wifi-connect/blob/master/scripts/raspbian-install.sh)
3. Run `sudo wifi-connect -p test123`

I did it in such a sequence, did not run other instances or something else

Result:
```
sudo wifi-connect -p test123
WiFi device: wlan0
Access points: ["GreatFruit", "YOTA", "GreatFruit", "OZPP", "GROUP", "sibalt", "Rosberry_2.4", "WiFi-DOM.ru-3060"]
Starting access point...
Access point 'WiFi Connect' created
Starting HTTP server on 192.168.42.1:80
[wifi_connect:ERROR] Cannot start HTTP server on '192.168.42.1:80': address not available
pi@raspberrypi:~ $ 
dnsmasq: unknown interface wlan0
```

**Comment by majorz** [comment]
@antonmolodykh oh, it is the passphrase that is too short that is causing the problem. It has to be 8 to 63 characters long. I will need to patch this properly.

**Comment by antonmolodykh** [comment]
@majorz  yes, it seems that this is the cause of the problem. Thank you

**Comment by majorz** [comment]
I will reopen the ticket so that we make sure the short passphrase issue gets fixed.

**Comment by majorz** [comment]
This can be worked on when https://github.com/resin-io-modules/network-manager/issues/111 is ready.

**Comment by lurch** [comment]
Clarified the title because I initially I thought it was talking about the range of ASCII code-points that were allowable in the password ;-)

**Comment by majorz** [comment]
:D

**Comment by tdoneal** [comment]
Bump.  I ran into this error too.  

It's confusing that "Error: Cannot start HTTP server address not available" is printed because of a short password.  If a long passphrase policy is to be enforced, it would be nice to see an error message along those lines.
